src/Service/PackageSynchronizer/ComposerPackageSynchronizer.php

```diff
@@ -207,4 +225,28 @@ private function createConfig(Package $package, IOInterface $io): Config
 
         return $config;
     }
+
+    private function loadREADME(Dist $dist): ?string
```

Comment:
I think this should be moved to a separate class, or to `Markdown` service. Maybe then you can name it `ReadmeExtractor` or something like that.

Reply:
Done.

As it's now its own class, it could be tested separately from the ComposerPackageSyncronizer. I think for now, we could leave the existing tests I added, as they are testing selecting the correct version.

If further readme files are added in the future, then they should be in a dedicated unit test.